depext.ml

```diff
@@ -161,7 +161,7 @@ let install_packages_commands ~interactive packages =
     else if distribution = "openbsd" then ["pkg_add"::yes ~no:["-i"] ["-I"] packages]
     else ["pkgin"::yes ["-y"] ("install"::packages)]
   | "archlinux" | "arch" ->
-    ["pacman"::"-S"::yes ["--noconfirm"] packages]
+    ["pacman"::"-Su"::yes ["--noconfirm"] packages]
```

Comment:
Along with upgrading, the latest version of the database should be used or the packages may not be available on the mirror anymore:
```suggestion
    ["pacman"::"-Syu"::yes ["--noconfirm"] packages]
```

Reply:
no, that's a separate option `opam depext -u` (which calls `pacman -Sy`)

Reply:
I see. The pacman manual recommends doing both because mirrors only have the latest version.

Reply:
indeed but it's the same thing with Debian, Alpine, NetBSD, OpenBSD at the very least. They all do the same thing of requiring the user to download explicitly.

Though it could be argued that since the rest of the package managers implicitly download the database, it would be interesting to unify the behaviours across platforms. Though this is outside of the scope of this particular change. I'll open an issue to track this separately.

Reply:
https://github.com/ocaml/opam/issues/4557

Reply:
I see the conundrum. With the appropriate flags arch could get the same behaviour as fedora. `pacman -Syu` only downloads a fresh copy if the local one is old enough.
To force downloading the database `-Syy` is needed.

Reply:
`pacman -Sy` vs. `pacman -Syy` isn't quite the same behaviour either.
The first one is "check the date of the remote database archive and download if the date doesn't match".
Whereas the second on is "force download without checking".

Homebrew and Fedora have a local timer which says "download when N days has passed since the last download", which is clearly different.